File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> release;
    int n = progresses.size();
    for (int i = 0; i < progresses.size(); i++) {

        if (((100 - progresses[i]) % speeds[i]) == 0) {
            release.push((100 - progresses[i]) / speeds[i]);
        }
        else {
            release.push((100 - progresses[i]) / speeds[i] + 1);
        }
    }
    while (!release.empty()) {
        int d = release.front(), p = 1;
        release.pop();

        while ((!release.empty()) && (release.front() <= d)) {
            release.pop();
            p++;
        }
        answer.push_back(p);
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import *
from collections import deque

def solution(progresses, speeds):
    daysLeft = list(map(lambda x: (ceil((100 - progresses[x]) / speeds[x])), range(len(progresses))))
    count = 1
    answer = []
    q = deque()
    q.append(daysLeft[0])
    idx = 1
    while len(q) != 0:
        v = q[0]
        try:
            if v >= daysLeft[idx]:
                q.append(daysLeft[idx])
            else:
                answer.append(len(q))
                q.clear()
                q.append(daysLeft[idx])
        except Exception:
            answer.append(len(q))
            break
        idx += 1

    return answer 